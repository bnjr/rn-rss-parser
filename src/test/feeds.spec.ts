import * as rssParser from '../index';

// Import feed XML samples directly as CommonJS
const reuters = require('./samples/reuters');

describe('when parse reuters', () => {
  describe('valid document', () => {
    it('should return feed items', async () => {
      const result = rssParser.parseFeed(reuters.feed);

      expect(result.title).toBe('Reuters: World News');
      expect(result.url).toBe('http://www.reuters.com');
      expect(result.description).toBe(
        "Reuters.com is your source for breaking news, business, financial and investing news, including personal finance and stocks.  Reuters is the leading global provider of news, financial information and technology solutions to the world's media, financial institutions, businesses and individuals.",
      );
      // Note: image property not currently implemented in model
      expect(result.language).toBe('en-us');
      expect(result.copyright).toBe(
        'All rights reserved. Users may download and print extracts of content from this website for their own personal and non-commercial use only. Republication or redistribution of Reuters content, including by framing or similar means, is expressly prohibited without the prior written consent of Reuters. Reuters and the Reuters sphere logo are registered trademarks or trademarks of the Reuters group of companies around the world. Â© Reuters 2018',
      );
      expect(result.items.length).toBe(10);
      expect(result.items[0].title).toBe(
        'U.S. general says North Korea not demonstrated all components of ICBM',
      );
      expect(result.items[0].categories?.length || result.items[0].categories.length).toBe(1);
      expect(result.items[0].categories[0].term).toBe('worldNews');
      expect(result.items[1].title).toBe(
        "Publication of Russia 'oligarch list' may affect investors: group",
      );
      expect(result.items[1].categories?.length || result.items[1].categories.length).toBe(1);
      expect(result.items[1].categories[0].term).toBe('worldNews');
      expect(result).toMatchSnapshot();
    });
  });
});
