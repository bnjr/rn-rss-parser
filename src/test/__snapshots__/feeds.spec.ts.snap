// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when parse reuters valid document should return feed items 1`] = `
{
  "authors": [],
  "categories": [
    {
      "label": "worldNews",
      "term": "worldNews",
      "url": null,
    },
    {
      "label": "worldNews",
      "term": "worldNews",
      "url": null,
    },
    {
      "label": "worldNews",
      "term": "worldNews",
      "url": null,
    },
    {
      "label": "worldNews",
      "term": "worldNews",
      "url": null,
    },
    {
      "label": "worldNews",
      "term": "worldNews",
      "url": null,
    },
    {
      "label": "worldNews",
      "term": "worldNews",
      "url": null,
    },
    {
      "label": "worldNews",
      "term": "worldNews",
      "url": null,
    },
    {
      "label": "worldNews",
      "term": "worldNews",
      "url": null,
    },
    {
      "label": "worldNews",
      "term": "worldNews",
      "url": null,
    },
    {
      "label": "worldNews",
      "term": "worldNews",
      "url": null,
    },
    {
      "label": "worldNews",
      "term": "worldNews",
      "url": null,
    },
  ],
  "copyright": "All rights reserved. Users may download and print extracts of content from this website for their own personal and non-commercial use only. Republication or redistribution of Reuters content, including by framing or similar means, is expressly prohibited without the prior written consent of Reuters. Reuters and the Reuters sphere logo are registered trademarks or trademarks of the Reuters group of companies around the world. Â© Reuters 2018",
  "description": "Reuters.com is your source for breaking news, business, financial and investing news, including personal finance and stocks.  Reuters is the leading global provider of news, financial information and technology solutions to the world's media, financial institutions, businesses and individuals.",
  "generator": null,
  "image": {
    "title": "Reuters News",
    "url": "http://www.reuters.com/resources_v2/images/reuters125.png",
  },
  "items": [
    {
      "authors": [],
      "categories": [
        {
          "label": "worldNews",
          "term": "worldNews",
          "url": "category",
        },
      ],
      "content": null,
      "description": "WASHINGTON (Reuters) - North Korea's nuclear program has made strides in recent months but the country has not yet demonstrated all the components of an intercontinental ballistic missile (ICBM), including a survivable re-entry vehicle, the vice chairman of the U.S. Joint Chiefs of Staff said on Tuesday.",
      "id": "http://www.reuters.com/article/us-northkorea-missiles-usa-idUSKBN1FJ2J0",
      "image": null,
      "media": [],
      "published": "2018-01-30T21:14:05.000Z",
      "title": "U.S. general says North Korea not demonstrated all components of ICBM",
      "updated": "2018-01-30T21:14:05.000Z",
      "url": "http://feeds.reuters.com/~r/Reuters/worldNews/~3/ilKUCdKYYyM/u-s-general-says-north-korea-not-demonstrated-all-components-of-icbm-idUSKBN1FJ2J0",
    },
    {
      "authors": [],
      "categories": [
        {
          "label": "worldNews",
          "term": "worldNews",
          "url": "category",
        },
      ],
      "content": null,
      "description": "MOSCOW (Reuters) - The U.S. Treasury Department's so-called "oligarchs' list" of Russian politicians and business leaders, compiled as part of a sanctions package signed into law last summer, may have consequences for listed persons even if it does not automatically trigger new economic restrictions.",
      "id": "http://www.reuters.com/article/us-usa-russia-sanctions-association-idUSKBN1FJ2FT",
      "image": null,
      "media": [],
      "published": "2018-01-30T19:31:53.000Z",
      "title": "Publication of Russia 'oligarch list' may affect investors: group",
      "updated": "2018-01-30T19:31:53.000Z",
      "url": "http://feeds.reuters.com/~r/Reuters/worldNews/~3/xSvw1-JuxFY/publication-of-russia-oligarch-list-may-affect-investors-group-idUSKBN1FJ2FT",
    },
    {
      "authors": [],
      "categories": [
        {
          "label": "worldNews",
          "term": "worldNews",
          "url": "category",
        },
      ],
      "content": null,
      "description": "MONROVIA (Reuters) - Liberian police fired live ammunition and tear gas on Tuesday to disperse a crowd trying to destroy a quarantine center on the outskirts of the capital, as a riot broke out in the seaside slum after its residents were told they had to stay there for 21 days.",
      "id": "http://www.reuters.com/article/us-liberia-police-idUSKBN1FJ2L4",
      "image": null,
      "media": [],
      "published": "2018-01-30T22:16:42.000Z",
      "title": "Shots fired as Liberia police battle illegal squatters",
      "updated": "2018-01-30T22:16:42.000Z",
      "url": "http://feeds.reuters.com/~r/Reuters/worldNews/~3/I1gqgDc6dlY/shots-fired-as-liberia-police-battle-illegal-squatters-idUSKBN1FJ2L4",
    },
    {
      "authors": [],
      "categories": [
        {
          "label": "worldNews",
          "term": "worldNews",
          "url": "category",
        },
      ],
      "content": null,
      "description": "WASHINGTON (Reuters) - Senior U.S. officials are debating whether they can legally and logistically carry out a limited strike on North Korea, the Wall Street Journal reported on Tuesday, as Washington seeks to deter Pyongyang's nuclear and missile development efforts.",
      "id": "http://www.reuters.com/article/us-northkorea-missiles-usa-strategy-idUSKBN1FJ365",
      "image": null,
      "media": [],
      "published": "2018-01-30T22:52:30.000Z",
      "title": "Trump officials debate targeted N. Korea strike in 'bloody nose' strategy: report",
      "updated": "2018-01-30T22:52:30.000Z",
      "url": "http://feeds.reuters.com/~r/Reuters/worldNews/~3/YCYDZZrjrZk/trump-officials-debate-targeted-n-korea-strike-in-bloody-nose-strategy-report-idUSKBN1FJ365",
    },
    {
      "authors": [],
      "categories": [
        {
          "label": "worldNews",
          "term": "worldNews",
          "url": "category",
        },
      ],
      "content": null,
      "description": "WASHINGTON (Reuters) - The U.S. State Department is reviewing whether a 2014 law banning defense sales to countries that use child soldiers applies to Saudi Arabia, the Pentagon said on Tuesday, in a move that comes as the two countries negotiate a huge weapons sale.",
      "id": "http://www.reuters.com/article/us-usa-canada-arms-idUSKBN1FJ333",
      "image": null,
      "media": [],
      "published": "2018-01-30T22:54:13.000Z",
      "title": "State Department reviewing whether Canada arms sales meet human rights test",
      "updated": "2018-01-30T22:54:13.000Z",
      "url": "http://feeds.reuters.com/~r/Reuters/worldNews/~3/XCmY5L3KYjo/state-department-reviewing-whether-canada-arms-sales-meet-human-rights-test-idUSKBN1FJ333",
    },
    {
      "authors": [],
      "categories": [
        {
          "label": "worldNews",
          "term": "worldNews",
          "url": "category",
        },
      ],
      "content": null,
      "description": "WASHINGTON (Reuters) - U.S. President Donald Trump is expected to ask for $716 billion for defense spending in the 2019 budget request he will send to Congress on Feb. 5, the Washington Post reported on Friday, citing two U.S. officials familiar with the matter.",
      "id": "http://www.reuters.com/article/us-usa-pentagon-budget-idUSKBN1FJ2ZA",
      "image": null,
      "media": [],
      "published": "2018-01-30T21:54:26.000Z",
      "title": "Trump plans to ask for $716 billion for defense in 2019 budget: Washington Post",
      "updated": "2018-01-30T21:54:26.000Z",
      "url": "http://feeds.reuters.com/~r/Reuters/worldNews/~3/WXxF3CFkZFU/trump-plans-to-ask-for-716-billion-for-defense-in-2019-budget-washington-post-idUSKBN1FJ2ZA",
    },
    {
      "authors": [],
      "categories": [
        {
          "label": "worldNews",
          "term": "worldNews",
          "url": "category",
        },
      ],
      "content": null,
      "description": "SANTIAGO (Reuters) - The sewers of Chilean capital Santiago have proven deadly for six young, renegade caimans, which were captured by animal control services after being flushed down the toilet, officials said on Tuesday.",
      "id": "http://www.reuters.com/article/us-chile-caimans-idUSKBN1FJ2HA",
      "image": null,
      "media": [],
      "published": "2018-01-30T20:38:24.000Z",
      "title": "Chile captures multiple crocodiles that went down drain",
      "updated": "2018-01-30T20:38:24.000Z",
      "url": "http://feeds.reuters.com/~r/Reuters/worldNews/~3/7hZqYD5ZZOI/chile-captures-multiple-crocodiles-that-went-down-drain-idUSKBN1FJ2HA",
    },
    {
      "authors": [],
      "categories": [
        {
          "label": "worldNews",
          "term": "worldNews",
          "url": "category",
        },
      ],
      "content": null,
      "description": "WASHINGTON (Reuters) - The United States is "extremely concerned" about reports that Syrian forces recently attacked citizens with chlorine gas, U.S. Secretary of State Rex Tillerson said on Tuesday.",
      "id": "http://www.reuters.com/article/us-mideast-crisis-syria-usa-idUSKBN1FJ2GT",
      "image": null,
      "media": [],
      "published": "2018-01-30T20:37:42.000Z",
      "title": "U.S. 'extremely concerned' about reports of chlorine gas use in Syria",
      "updated": "2018-01-30T20:37:42.000Z",
      "url": "http://feeds.reuters.com/~r/Reuters/worldNews/~3/gfJQF8eElpg/u-s-extremely-concerned-about-reports-of-chlorine-gas-use-in-syria-idUSKBN1FJ2GT",
    },
    {
      "authors": [],
      "categories": [
        {
          "label": "worldNews",
          "term": "worldNews",
          "url": "category",
        },
      ],
      "content": null,
      "description": "WASHINGTON (Reuters) - The Central Intelligence Agency believes North Korea's missile program is aimed at coercion, not just self-defense, and that Pyongyang's next logical step would be to develop an arsenal of weapons and the capability to fire multiple missiles at the United States, CIA Director Mike Pompeo said on Tuesday.",
      "id": "http://www.reuters.com/article/us-northkorea-missiles-usa-cia-idUSKBN1FJ2QN",
      "image": null,
      "media": [],
      "published": "2018-01-30T20:19:48.000Z",
      "title": "CIA says believes North Korea weapons aimed at coercion, not just defense: Pompeo",
      "updated": "2018-01-30T20:19:48.000Z",
      "url": "http://feeds.reuters.com/~r/Reuters/worldNews/~3/--iJfEA4h6I/cia-says-believes-north-korea-weapons-aimed-at-coercion-not-just-defense-pompeo-idUSKBN1FJ2QN",
    },
    {
      "authors": [],
      "categories": [
        {
          "label": "worldNews",
          "term": "worldNews",
          "url": "category",
        },
      ],
      "content": null,
      "description": "GAZA (Reuters) - The Islamist group Hamas has handed control of the Gaza Strip's border crossings with Israel and Egypt to the Palestinian Authority, the group said on Wednesday.",
      "id": "http://www.reuters.com/article/us-palestinians-gaza-idUSKBN1FJ2FY",
      "image": null,
      "media": [],
      "published": "2018-01-30T19:33:25.000Z",
      "title": "Hamas says it hands over control of Gaza border crossings",
      "updated": "2018-01-30T19:33:25.000Z",
      "url": "http://feeds.reuters.com/~r/Reuters/worldNews/~3/QnmGJLwCYVo/hamas-says-it-hands-over-control-of-gaza-border-crossings-idUSKBN1FJ2FY",
    },
  ],
  "language": "en-us",
  "meta": {
    "type": "rss",
    "version": "2.0",
  },
  "published": "2018-01-30T21:14:05.000Z",
  "self": null,
  "title": "Reuters: World News",
  "updated": "2018-01-30T22:54:13.000Z",
  "url": "http://www.reuters.com",
}
`;
